bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/handler/MeteoAlerteHandler.java

```diff
@@ -133,74 +136,55 @@ private void updateAndPublish() {
                 }));
     }
 
-    public void updateIcon(String channelId, String value) {
-        String iconChannelId = channelId + "-icon";
-        if (isLinked(iconChannelId)) {
-            String pictoName = channelId + (!value.isEmpty() ? "_" + value.toLowerCase() : "");
-            byte[] image = getImage("picto" + File.separator + pictoName + ".gif");
-            if (image != null) {
-                RawType picto = new RawType(image, "image/gif");
-                updateState(iconChannelId, picto);
-            }
-        }
-    }
-
-    private byte @Nullable [] getImage(String iconPath) {
-        URL url = FrameworkUtil.getBundle(getClass()).getResource(iconPath);
-        logger.debug("Path to icon image resource is: {}", url);
-        try (InputStream in = new BufferedInputStream(url.openStream())) {
-            ByteArrayOutputStream bos = new ByteArrayOutputStream();
-            int next = in.read();
-            while (next > -1) {
-                bos.write(next);
-                next = in.read();
-            }
-            bos.flush();
-            return bos.toByteArray();
+    public @Nullable String getResource(String iconPath) {
+        Bundle bundle = FrameworkUtil.getBundle(getClass());
+        try (InputStream stream = bundle.getResource(iconPath).openStream()) {
+            return new BufferedReader(new InputStreamReader(stream)).lines().collect(Collectors.joining("\n"));
         } catch (IOException e) {
-            logger.debug("I/O exception occurred getting image data: {}", e.getMessage(), e);
+            logger.warn("Unable to load ressource '{}' : {}", iconPath, e.getMessage());
         }
         return null;
     }
 
-    public void updateAlertString(String channelId, String value) {
-        if (!value.isEmpty() && isLinked(channelId)) {
-            int level = ALERT_LEVELS.indexOf(value);
-            if (level != -1) {
-                updateState(channelId, new StringType(Integer.toString(level)));
-            } else {
-                updateState(channelId, UnDefType.UNDEF);
-                logger.warn("Value {} is not a valid alert level for channel {}", value, channelId);
+    public void updateAlert(String channelId, AlertLevel value) {
+        String channelIcon = channelId + "-icon";
+        if (value != AlertLevel.UNKNOWN) {
+            if (isLinked(channelId)) {
+                updateState(channelId, new StringType(value.name()));
+            }
+            if (isLinked(channelIcon)) {
+                String resource = getResource(String.format("picto/%s.svg", channelId));
+                if (resource != null) {
+                    resource = resource.replaceAll(ALERT_COLORS.get(AlertLevel.UNKNOWN), ALERT_COLORS.get(value));
+                }
+                updateState(channelIcon,
+                        resource != null ? new RawType(resource.getBytes(), "image/svg+xml") : UnDefType.NULL);
```

Comment:
Rather UNDER?

Reply:
UNDEF / NULL... I always have this debate :-) I understood that UNDEF was reserved to the core prior any binding update.

Reply:
That is the reverse. 
NULL is the initial value and should be set only by the framework.